components/keys/src/time.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::fmt;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct TimeStamp(u64);
+
+const TSO_PHYSICAL_SHIFT_BITS: u64 = 18;
+
+impl TimeStamp {
+    /// Create a time stamp from physical and logical components.
+    pub fn compose(physical: u64, logical: u64) -> TimeStamp {
+        TimeStamp((physical << TSO_PHYSICAL_SHIFT_BITS) + logical)
+    }
+
+    pub const fn min() -> TimeStamp {
+        TimeStamp(0)
+    }
+
+    pub const fn max() -> TimeStamp {
+        TimeStamp(std::u64::MAX)
+    }
+
+    pub const fn new(ts: u64) -> TimeStamp {
+        TimeStamp(ts)
+    }
+
+    /// Extracts physical part of a timestamp, in milliseconds.
+    pub fn extract_physical(self) -> u64 {
+        self.0 >> TSO_PHYSICAL_SHIFT_BITS
+    }
+
+    pub fn incr(self) -> TimeStamp {
+        TimeStamp(self.0 + 1)
+    }
+
+    pub fn decr(self) -> TimeStamp {
+        TimeStamp(self.0 - 1)
+    }
+
+    pub fn is_zero(self) -> bool {
+        self.0 == 0
+    }
+
+    pub fn into_inner(self) -> u64 {
```

Comment:
`into_inner` is a pretty standard name for converting a wrapper type into the underlying type. I don't like to use `into_T` methods, I prefer implementing `Into<T>` which I considered here, but I thought it better to emphasise the unwrapping rather than the conversion. Why do you prefer `into_u64`?